input/system/azure/system.go

```diff
@@ -0,0 +1,302 @@
+							NodeServerEdition:        util.StringPtrToString(v.Properties.NodeServerEdition),
+							NodeCount:                util.Int32PtrToInt(v.Properties.NodeCount),
+						}
+						resourceID = *v.ID
+						break
+					}
+				}
+			}
+		}
+	}
+
+	if resourceID == "" {
+		server.SelfTest.MarkCollectionAspectWarning(state.CollectionAspectSystemStats, "unable to find the database server info")
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the Reader permission of the database is granted to the managed identity.")
+		return
+	}
+
+	// Server metrics
+	client, err := azquery.NewMetricsClient(credential, nil)
+	if err != nil {
+		server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error making a metrics client: %v", err)
+		logger.PrintError("Azure/System: Failed to make a metrics client: %v\n", err)
+		return
+	}
+
+	// Query metrics data with 1 min interval, for the last 1 min (should return 1 value)
+	metricNames := "cpu_percent,memory_percent,network_bytes_egress,network_bytes_ingress,storage_used"
+	if strings.ToLower(system.Info.Azure.ResourceType) == "flexibleserver" {
+		// metrics only available with Flexible Server
+		metricNames += ",txlogs_storage_used,read_iops,write_iops,disk_queue_depth,read_throughput,write_throughput"
+	}
+	option := &azquery.MetricsClientQueryResourceOptions{
+		MetricNames: to.Ptr(metricNames),
+		Aggregation: to.SliceOfPtrs(azquery.AggregationTypeAverage),
+		Interval:    to.Ptr("PT1M"),
+		Timespan:    to.Ptr(azquery.TimeInterval("PT1M")),
+	}
+
+	metricsRes, err := client.QueryResource(ctx, resourceID, option)
+	if err != nil {
+		server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error getting server metrics: %v", err)
+		logger.PrintError("Azure/System: Encountered error getting server metrics: %v\n", err)
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the Monitoring Reader permission of the database is granted to the managed identity.")
+		return
+	}
+
+	system.CPUStats = make(state.CPUStatisticMap)
+	diffedNetworkStats := &state.DiffedNetworkStats{}
+	diffedDiskStats := &state.DiffedDiskStats{}
+	for _, metric := range metricsRes.Value {
+		if *metric.Name.Value == "cpu_percent" {
+			// Should be only one data as 1 min time span with 1 min interval is selected
+			for _, timeSeriesElement := range metric.TimeSeries {
+				for _, metricValue := range timeSeriesElement.Data {
+					if metricValue.Average != nil {
+						system.CPUStats["all"] = state.CPUStatistic{
+							DiffedOnInput: true,
+							DiffedValues: &state.DiffedSystemCPUStats{
+								UserPercent: *metricValue.Average,
+							},
+						}
+					}
+				}
+			}
+		} else if *metric.Name.Value == "memory_percent" {
+			// Currently, we can only retrieve memory percent
+			// Since total memory size is not listed in the server info,
+			// we are unable to pass this value to pganalyze at the moment
```

Comment:
While the `memory_percent` metrics is available for both Flexible Server and Cosmos DB, it's super hard to get the memory size information of the server. Since pganalyze doesn't accept the memory percentage at the moment, I simply skipped recording this for now.

In the future, we could try obtaining the memory information of the server somehow. One option is to have our own SKU and memory size mapping data here (the list of sku can be obtained via `az postgres flexible-server list-skus --location eastus -o table` command), but looks like there is no easy place to find that information with Cosmos DB (the same CLI command is not available for Cosmos DB), so I decided to skip it.